### PR TITLE
Upgrades to support multi readout charge collection

### PR DESCRIPTION
**Pull request by jgalan**
Large jgalan  [](https://gitlab.cern.ch/rest-for-physics/detectorlib/-/commits/jgalan_segmented_mesh)

I had to update `TRestDetectorSignalToHitsProcess` so that it is capable to add charge to more than one `TRestDetectorReadoutPlane`. In general it will add charge to any readout plane definition where the hit is found inside the volume defined by cathode and readout plane positions (and the charge falls inside a readout channel).

List of changes:
- `TRestDetectorSignalToHitsProcess` has now been fully documented. Including image showing the process input and output event. https://sultan.unizar.es/test/classTRestDetectorHitsToSignalProcess.html

- `TRestDetectorReadout::GetHitsDaqChannelAtReadoutPlane` method added to check the *daqId* for a specific readout plane so that `TRestDetectorHitsToSignalProcess` iterates each plane and gets the signal id associated with each plane.

-  `TRestDetectorHitsEvent::DrawEvent` aesthetical improvements. And a fix.

- `TRestDetectorHitsToSignalProcess` now returns `nullptr` if no signals have been found. The event will then be ignored. 

This update was required by segmented mesh microbulks.

### CODE REVIEW

File: inc/TRestDetectorReadout.h [hunk 1]
```diff
@@ -78,6 +78,8 @@ class TRestDetectorReadout : public TRestMetadata {
     TRestDetectorReadoutModule* ParseModuleDefinition(TiXmlElement* moduleDefinition);
     void GetPlaneModuleChannel(Int_t daqID, Int_t& planeID, Int_t& moduleID, Int_t& channelID);
     Int_t GetHitsDaqChannel(TVector3 hitpos, Int_t& planeID, Int_t& moduleID, Int_t& channelID);
+    Int_t GetHitsDaqChannelAtReadoutPlane(TVector3 hitpos, Int_t& moduleID, Int_t& channelID,
```
Comment (juanangp) [inline inc/TRestDetectorReadout.h:81]: Please pass args as reference whenever is possible:
```
 Int_t GetHitsDaqChannelAtReadoutPlane(TVector3 hitpos, Int_t& moduleID, Int_t& channelID,Int_t planeId = 0);
-->
 Int_t GetHitsDaqChannelAtReadoutPlane(const TVector3 &hitpos, Int_t& moduleID, Int_t& channelID,Int_t planeId = 0);
```

File: src/TRestDetectorHitsToSignalProcess.cxx [hunk 2]
```diff
@@ -32,13 +124,18 @@ TRestDetectorHitsToSignalProcess::TRestDetectorHitsToSignalProcess(char* cfgFile
...
 TRestDetectorHitsToSignalProcess::~TRestDetectorHitsToSignalProcess() {
     if (fReadout != NULL) delete fReadout;
 
     delete fSignalEvent;
 }
 
+///////////////////////////////////////////////
+/// \brief Function to load the default config in absence of RML input
+///
 void TRestDetectorHitsToSignalProcess::LoadDefaultConfig() {
```
Comment (juanangp) [inline src/TRestDetectorHitsToSignalProcess.cxx:139]: Don´t quite understand this funcion, variables have been already initialized in the header

File: src/TRestDetectorReadout.cxx [hunk 3]
```diff
@@ -768,6 +768,45 @@ Int_t TRestDetectorReadout::GetHitsDaqChannel(TVector3 hitpos, Int_t& planeID, I
...
+                                                            Int_t& channelID, Int_t planeId) {
+    if (planeId > GetNumberOfReadoutPlanes()) {
+        warning << "TRestDetectorReadout. Fail trying to retrieve planeId : " << planeId << endl;
+        warning << "Number of readout planes: " << GetNumberOfReadoutPlanes() << endl;
+        return -1;
+    }
+
+    double x = hitpos.X();
+    double y = hitpos.Y();
+    double z = hitpos.Z();
```
Comment (juanangp) [inline src/TRestDetectorReadout.cxx:794]: Doesn't looks like these variables are used at all, remove them please

File: src/TRestDetectorHitsToSignalProcess.cxx (hunk 2 above)
Comment (jgalan) [reply src/TRestDetectorHitsToSignalProcess.cxx:141]: Right, that is the new scheme. Probably now is useless and should be removed. Likely a different issue for this PR.

**Review by juanangp — APPROVED** [review]

**Pull request merged**